### PR TITLE
Add envoy metrics service configuration to ProxyConfig

### PR DESCRIPTION
**Pull request by joeyb**
Similar to the statsd sink config, this change is in preparation for adding support for the envoy metrics service API.

### CODE REVIEW

File: mesh/v1alpha1/proxy.proto [hunk 1]
```diff
@@ -164,4 +164,7 @@ message ProxyConfig {
 
   // Tracing configuration to be used by the proxy.
   Tracing tracing = 19;
+
+  // Address of the Envoy Metrics Service implementation (e.g. metrics-service:15000).
+  string metrics_service_address = 20;
```
Comment (rshriram) [inline mesh/v1alpha1/proxy.proto]: can you add this below statsd_udp_address ?
And is this mutually exclusive with statsd or can both co-exist?

Comment (joeyb) [reply mesh/v1alpha1/proxy.proto]: > can you add this below statsd_udp_address ?

👍 

> And is this mutually exclusive with statsd or can both co-exist?

Both should be able to co-exist. Here's a draft of the changes in istio/istio: https://github.com/istio/istio/pull/11897

**Comment by joeyb** [comment]
@rshriram - Just added a draft of the istio/istio PR: https://github.com/istio/istio/pull/11897

**Comment by rshriram** [comment]
merging. adding API field to proxy config with no backward compat issues..

**Comment by rshriram** [comment]
could you complete your PR in istio/istio ?

**Pull request merged**